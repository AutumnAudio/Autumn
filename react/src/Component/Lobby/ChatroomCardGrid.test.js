import React from 'react'
import renderer from 'react-test-renderer'
import ChatroomCardGrid from './ChatroomCardGrid'

test('should be able to render chatroom card grid', () => {
    const chatroomsMock = [{chat: [],
                          genre: 'JAZZ',
                          link: '/joinroom/JAZZ',
                          participants: {'user1': {username: 'user1'}},
                          playlist: []},
                          {chat: [],
                            genre: 'POP',
                            link: '/joinroom/POP',
                            participants: {'user2': {username: 'user2'}},
                            playlist: []}]
    const component = renderer.create(
        <ChatroomCardGrid chatrooms={chatroomsMock}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
