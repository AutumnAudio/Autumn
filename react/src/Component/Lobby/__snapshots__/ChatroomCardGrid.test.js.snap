// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be able to render chatroom card grid 1`] = `
<div>
  <div
    className="MuiGrid-root ChatroomCardGrid-root-1 MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-6 MuiGrid-justify-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <div
            className="MuiPaper-root ChatroomCard-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
            onClick={[Function]}
          >
            <div>
              Online: 
              1
            </div>
            <div
              className="ChatroomCard-chatroomName-3"
            >
              JAZZ
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <div
            className="MuiPaper-root ChatroomCard-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
            onClick={[Function]}
          >
            <div>
              Online: 
              1
            </div>
            <div
              className="ChatroomCard-chatroomName-3"
            >
              POP
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
